### PR TITLE
Fix single frame render not publishing correctly

### PR DESCRIPTION
**Pull request by jpbrault**
JIRA: SMOK-48860
Can't Publish 1 frame to Shotgun from Batch.

The regex used for finding the frame range was considering single frame
pattern but was not extracting the information correctly.

### CODE REVIEW

File: hooks/tk-multi-publish2/collector.py [hunk 1]
```diff
@@ -792,20 +795,23 @@ def _get_file_sequence(self, path):
...
+            return [] # The path is not a sequence
 
         # Get the first and last frame of the sequence
-        first, last = match.group(2).replace("[", "").replace("]", "").split("-")
+        frame_range = match.group(2).replace("[", "").replace("]", "").split("-")
+        if len(frame_range) == 1:
+            last = first = frame_range[0]
+        elif len(frame_range) == 2:
+            first, last = frame_range
+        else:
```
Comment (geffrak) [inline hooks/tk-multi-publish2/collector.py]: Unreachable right ?

Comment (jpbrault) [reply hooks/tk-multi-publish2/collector.py]: Right. Should not be possible with regex.

File: hooks/tk-multi-publish2/collector.py [hunk 2]
```diff
@@ -792,20 +795,21 @@ def _get_file_sequence(self, path):
...
-            return file_sequence
+            return [] # The path is not a sequence
 
         # Get the first and last frame of the sequence
-        first, last = match.group(2).replace("[", "").replace("]", "").split("-")
+        frame_range = match.group(2).replace("[", "").replace("]", "").split("-")
+        if len(frame_range) == 1:
+            last = first = frame_range[0]
+        else:
+            first, last = frame_range
```
Comment (manneohrstrom) [inline hooks/tk-multi-publish2/collector.py]: can frame range be more than 2 items?

Comment (jpbrault) [reply hooks/tk-multi-publish2/collector.py]: The regex should prevent it as Kerby pointed out.

**Pull request merged**